File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = float('-inf')

        current_sum = 0
        for i in range(len(nums)):
            current_sum = max(nums[i], nums[i] + current_sum)
            max_sum = max(max_sum, current_sum)

        return max_sum

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {

        int max_sum = Integer.MIN_VALUE;
        int current_sum = 0;

        for (int i = 0; i < nums.length; i++) {
            current_sum = Math.max(nums[i], nums[i] + current_sum);
            max_sum = Math.max(current_sum, max_sum);
        }

        return max_sum;
        
    }
}